core/trino-main/src/main/java/io/trino/sql/analyzer/Analysis.java

```diff
@@ -844,6 +845,11 @@ public void addReferencedFields(Multimap<NodeRef<Node>, Field> references)
         referencedFields.putAll(references);
     }
 
+    public void addReferencedFields(Multimap<NodeRef<Node>, Field> references, Expression expression)
```

Comment:
Adjust the method name.
BTW, I can see no usage of the field `referencedFields` in `Analysis`. If you removed it, you could simplify creating `originColumnDetails` by avoiding one level of mapping by the source node.